### PR TITLE
HPCC-17158 Improve Sort error emanating from slave

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:

Observed and tested when fixing HPCC-17155

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

**Comment by richardkchapman** [comment]
Typo in commit message (should be emanating)

File: thorlcr/msort/tsortmp.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ bool SortSlaveMP::sendRecv(CMessageBuffer &mb, unsigned timeout)
             mb.read(err);
             StringAttr errstr;
             mb.read(errstr);
-            throw MakeStringException(err, "%s", errstr.get());
+            throw MakeStringException(err, "SortSlaveMP(error from slave %u) - %s", (unsigned)rank, errstr.get());
```
Comment (richardkchapman) [inline thorlcr/msort/tsortmp.cpp]: That may be more meaningful to you, but is it really much more meaningful to anyone else that might see it? What does SortSlaveMP mean ?

Comment (jakesmith) [inline thorlcr/msort/tsortmp.cpp]: It was the addition of slave number that was the significant new piece of information .
I'll replace SortGlobalMP with something else.

**Comment by jakesmith** [comment]
@richardkchapman - changed to use std. activity exception format, please recheck.

File: thorlcr/msort/tsortm.cpp [hunk 2]
```diff
@@ -90,18 +90,14 @@ class CSortNode: public SortSlaveMP
     bool            overflow;
     unsigned        scale;     // num times overflowed
     
-    CActivityBase *activity;
-    
-    
-    CSortNode(CActivityBase *_activity, ICommunicator *comm,rank_t _rank, SocketEndpoint &sep,mptag_t _mpTagRPC, CSortMaster &_sorter) 
-        : sorter(_sorter)
+    CSortNode(CActivityBase *activity, ICommunicator *comm,rank_t _rank, SocketEndpoint &sep,mptag_t _mpTagRPC, CSortMaster &_sorter)
+        : SortSlaveMP(activity), sorter(_sorter)
```
Comment (mckellyln) [inline thorlcr/msort/tsortm.cpp]: trivial, is the pattern still a leading underbar, as in _activity ?

**Review by mckellyln — APPROVED** [review]
tirivial comment, ok to merge code.

**Pull request merged**